File: crates/domain/src/media/subtitle_candidate.rs
use crate::common::LanguageCode;
use crate::media::SubtitleFormat;

#[derive(Debug, Clone, PartialEq)]
pub struct SubtitleCandidate {
    pub file_id: String,
    pub language: Option<LanguageCode>,
    pub format: SubtitleFormat,
    pub release_name: Option<String>,
    pub download_count: Option<u32>,
    pub rating: Option<f32>,
}

File: clients/flutter/lib/model/catalog/subtitle_candidate.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import 'package:shadowmask/model/catalog/version_detail.dart';

part 'subtitle_candidate.freezed.dart';
part 'subtitle_candidate.g.dart';

@freezed
abstract class SubtitleCandidate with _$SubtitleCandidate {
  const factory SubtitleCandidate({
    required String fileId,
    String? language,
    String? releaseName,
    required SubtitleFormat format,
    int? downloadCount,
    double? rating,
  }) = _SubtitleCandidate;

  factory SubtitleCandidate.fromJson(Map<String, dynamic> json) =>
      _$SubtitleCandidateFromJson(json);
}
